### DISCUSSION TITLE
classes/class_collisionobject3d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#135] [User-contributed notes]
# classes/class_collisionobject3d

Inherits: Node3D< Node< Object Inherited By: Area3D, PhysicsBody3D Abstract base class for 3D physics objects. Description: Abstract base class for 3D physics objects. CollisionObject3D can hold an...

https://docs.godotengine.org/en/stable/classes/class_collisionobject3d.html

**Comment by Cirrial** [comment]
input_ray_pickable does not function if the mouse is captured by the screen, and after trying to investigate further this appears to be by design. You'll have to set up your own raycasts if you need functionality like it while the mouse is captured.

**Reply by Juanxpeke** [reply]
```input_ray_pickable``` and all the related signals / methods only works when ```Settings:physics/common/enable_object_picking``` or ```Viewport.physics_object_picking``` are enabled.

**Comment by Juanxpeke** [comment]
```input_ray_pickable``` and all the related signals / methods only work when ```Settings:physics/common/enable_object_picking``` or ```Viewport.physics_object_picking``` are enabled.

**Comment by ArizoftGames** [comment]
It looks like the signal params don't match the expected params for _InputEvent; which fails any override method.  (Node: camera vs Camera3D: camera) The workaround I'm using is to create a separate OnInputEvent method which expects the same params passed by the signal.  Is this mismatch intentional, or is there something I'm missing?  Can't seem to find any other references to this.
